src/lib.rs

```diff
@@ -121,22 +125,33 @@ impl std::fmt::Display for Program {
     }
 }
 
-#[derive(Debug)]
+#[derive(Debug, PartialEq)]
 pub struct CompileError {
     cmds: Vec<Cmd>,
-    index: usize,
-    stack_size: isize,
+    error_kind: ErrorKind,
```

Comment:
We should have a way for people to observe the error kind so they can handle it besides just printing it. Either we should make this field public, or (more likely) we should have a method that gives a reference to it.